presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -139,12 +139,24 @@ public StatisticRange intersect(StatisticRange other)
         return empty();
     }
 
-    public StatisticRange add(StatisticRange other)
+    public StatisticRange addAndSumDistinctValues(StatisticRange other)
     {
         double newDistinctValues = distinctValues + other.distinctValues;
         return new StatisticRange(minExcludeNaN(low, other.low), maxExcludeNaN(high, other.high), newDistinctValues);
     }
 
+    public StatisticRange addAndCollapseDistinctValues(StatisticRange other)
+    {
+        double overlapPercentOfLeft = this.overlapPercentWith(other);
+        double overlapPercentOfRight = other.overlapPercentWith(this);
+        double overlapDistinctValuesLeft = overlapPercentOfLeft * distinctValues;
+        double overlapDistinctValuesRight = overlapPercentOfRight * other.distinctValues;
+        double maxOverlappingValues = maxExcludeNaN(overlapDistinctValuesLeft, overlapDistinctValuesRight);
```

Comment:
do we need `ExcludeNaN` here. In the next line we perform normal arithmetic without any NaN checking. 
Is that possible that current code will compute non-NaN value of `newDistinctValues` in case where normal `max(overlapDistinctValuesLeft, overlapDistinctValuesRight)` would compute `NaN`